extensions/vercast/src/pages/lists/menubar-deployments.tsx

```diff
@@ -0,0 +1,162 @@
+import { Icon, Color, MenuBarExtra, open, Keyboard } from "@raycast/api";
+import useVercel from "../../hooks/use-vercel-info";
+import fromNow from "../../utils/time";
+import { Deployment, DeploymentState } from "../../types";
+import { FetchHeaders, getFetchDeploymentsURL } from "../../vercel";
+import { useFetch } from "@raycast/utils";
+import MenuBarTeamAccessory from "../search-projects/team-switch-menubar-accessory";
+
+interface StateStyles extends Record<DeploymentState, { icon: string; color: string }> {
+  undefined: { icon: string; color: string };
+}
+
+const stateStyles: StateStyles = {
+  READY: {
+    icon: Icon.CheckCircle,
+    color: Color.Green,
+  },
+  BUILDING: {
+    icon: Icon.Bolt,
+    color: Color.Yellow,
+  },
+  INITIALIZING: {
+    icon: Icon.BoltDisabled,
+    color: Color.Yellow,
+  },
+  FAILED: {
+    icon: Icon.ExclamationMark,
+    color: Color.Red,
+  },
+  CANCELED: {
+    icon: Icon.MinusCircleFilled,
+    color: Color.Red,
+  },
+  ERROR: {
+    icon: Icon.Warning,
+    color: Color.Red,
+  },
+  QUEUED: {
+    icon: Icon.List,
+    color: Color.Purple,
+  },
+  undefined: {
+    icon: Icon.Dot,
+    color: Color.PrimaryText,
+  },
+};
+
+const seperator = "·";
+const newline = "\n";
+const numShow = 12;
+
+const MenuBarDeployments = ({ projectId }: { projectId?: string }) => {
+  const { user, selectedTeam } = useVercel();
+  const url = getFetchDeploymentsURL(selectedTeam?.id, projectId);
+
+  const { isLoading, data, revalidate } = useFetch<{
+    deployments: Deployment[];
+    // TODO: why can't I `{ headers: FetchHeaders }` here?
```

Comment:
I have no idea. Maybe @MaxLeiter or @gleich know?

Reply:
Sorry, I definitely wrote that line but can't remember what the actual typing error was.